File: assets/handlers/python/user_create/src/create_user/create.py
from create_user import db, unmarshal


async def create_user(json_data: bytes) -> str:
    user = unmarshal.unmarshal_user(json_data)
    insert_dict = user.model_dump()
    insert_result = await db.collection.insert_one(insert_dict)
    return str(insert_result.inserted_id)

File: assets/handlers/go/user_create/src/create/create.go
package create

import (
	"context"
	"user_create/src/db"
	"user_create/src/models"
	"user_create/src/unmarshal"
)

// Creates a new user in Database.
func CreateUser(jsonData []byte) (string, error) {
	// Unmarshal JSON data into a User struct
	userStruct, err := unmarshal.UnmarshalUser(jsonData)
	if err != nil {
		return "", err
	}

	user, _ := models.NewUser(userStruct)

	insertResult, err := db.UserCollection.InsertOne(context.Background(), user)
	if err != nil {
		return "", err
	}

	// Return the inserted ID as string
	return insertResult.InsertedID.(string), nil
}
